### ISSUE TITLE
api/vmagent: support flush_on_shutdown for streamingaggregation configuration

### ISSUE THREAD

**Issue by rjrooke** [VictoriaMetrics/operator#860]
It appears this configuration item may not be supported by the operator:

```
  # staleness_interval defines an interval after which the series state will be reset if no samples have been sent during it.
  # It means that:
  # - no data point will be written for a resulting time series if it didn't receive any updates during configured interval,
  # - if the series receives updates after the configured interval again, then the time series will be calculated from the initial state
  #   (it's like this series didn't exist until now).
  # Increase this parameter if it is expected for matched metrics to be delayed or collected with irregular intervals exceeding the `interval` value.
  # By default, is equal to x2 of the `interval` field.
  # The parameter is only relevant for outputs: total, increase and histogram_bucket.
  #
  # staleness_interval: 2m

```
https://docs.victoriametrics.com/stream-aggregation/#stream-aggregation-config

Error from server (BadRequest): error when replacing "webscale-vmb-feed.yaml": VMAgent in version "v1beta1" cannot be handled as a VMAgent: strict decoding error: unknown field "spec.remoteWrite[5].streamAggrConfig.rules[0].flush_on_shutdown"

**Comment by Haleygo** [comment]
Hello!
`flush_on_shutdown` is [not released yet](https://docs.victoriametrics.com/changelog/#tip) and will be included in the next release[v1.97.0].
And operator will add this field afterwards.

**Comment by f41gh7** [comment]
Added to the [v0.42.0 release](https://github.com/VictoriaMetrics/operator/releases/tag/v0.42.0)
